src/components/VmDetails/cards/SnapshotsCard/index.js

```diff
@@ -1,34 +1,46 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
+import { connect } from 'react-redux'
+
 import BaseCard from '../../BaseCard'
 import style from './style.css'
 
 import NewSnapshotModal from './NewSnapshotModal'
 import SnapshotItem from './SnapshotItem'
+import { PendingTaskTypes } from '../../../../reducers/pendingTasks'
 
-const Snapshots = ({ snapshots, vmId }) => {
+const Snapshots = ({ snapshots, vmId, beingCreated, beingRestored }) => {
   const isVmInPreview = !!snapshots.find(snapshot => snapshot.get('status') === 'in_preview')
   return (
     <React.Fragment>
+      <div className={style['snapshot-create']}>
+        <NewSnapshotModal vmId={vmId} disabled={isVmInPreview || beingCreated || beingRestored} />
+      </div>
       {
-        !isVmInPreview &&
-        <div className={style['snapshot-create']}>
-          <NewSnapshotModal vmId={vmId} />
-        </div>
+        snapshots.sort((a, b) => b.get('date') - a.get('date')).map((snapshot) => (
+          <SnapshotItem key={snapshot.get('id')} snapshot={snapshot} vmId={vmId} isEditing={!isVmInPreview && !beingRestored} />
+        ))
       }
-      {snapshots.sort((a, b) => b.get('date') - a.get('date')).map((snapshot) => (
-        <SnapshotItem key={snapshot.get('id')} snapshot={snapshot} vmId={vmId} isEditing={!isVmInPreview} />
-      ))}
     </React.Fragment>
   )
 }
 
 Snapshots.propTypes = {
   snapshots: PropTypes.object.isRequired,
   vmId: PropTypes.string.isRequired,
+  beingCreated: PropTypes.bool,
+  beingRestored: PropTypes.bool,
 }
 
+const ConnectedSnapshots = connect(
+  (state) => ({
+    beingCreated: !!state.pendingTasks.find(t => t.type === PendingTaskTypes.SNAPSHOT_ADD),
+    beingRestored: !!state.pendingTasks.find(task => task.type === PendingTaskTypes.SNAPSHOT_RESTORE),
+  }),
+  (dispatch) => ({})
```

Comment:
You can leave the `dispatch` section out completely.  `connect` can be called with a single argument.